### DISCUSSION TITLE
Question: How to implement the limit_except directive?

### DISCUSSION THREAD

**Discussion by tracure1337** [evertramos/nginx-proxy-automation#317] [Q&A] [answered]
I have successfully implement basic auth on my main route / for my application by adding the data/htpasswd  file named ```<FQDN>```.

Now I want to add a limit_except directive like:

```
cat <FDQN>_location
```
```
location /api {
    proxy_pass <FQDN>;

    limit_except POST {
        basic auth "off";
        }
}
```
Basically any route any method should ask for auth with exception of POST /api

I tried quite a few different solutions but none of them worked.

I hoped I can find an answer here ;)

**Comment by evertramos** [comment, marked as answer]
Hello @cocoonkid!

Thanks for you question.

Here is some help how you may customize the nginx-proxy or even only a few hosts, follow the link below to the nginx-proxy documentation:

[https://github.com/nginx-proxy/nginx-proxy#custom-nginx-configuration](https://github.com/nginx-proxy/nginx-proxy#custom-nginx-configuration)

Hope it helps!

**Reply by tracure1337** [reply]
Thank you!
